src/pylint/rules/magic_value_comparison.rs

```diff
@@ -0,0 +1,41 @@
+use rustpython_ast::{Constant, Expr, ExprKind};
+
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::registry::Diagnostic;
+use crate::violations;
+
+fn is_magic_value(constant: &Constant) -> Option<String> {
+    match constant {
+        Constant::Int(value) => {
+            if matches!(value.try_into(), Ok(-1 | 0 | 1)) {
+                None
+            } else {
+                Some(value.to_string())
+            }
+        }
+        Constant::Float(value) => Some(value.to_string()),
```

Comment:
It looks like Pylint flags on any constant, but allows strings `""` and `"__main__"`.

https://pylint.pycqa.org/en/latest/user_guide/configuration/all-options.html#valid-magic-values

Reply:
Have updated to reflect this, required a refactor of some of the workings but I believe for the better.